modules/core/04-channel/types/channel.go

```diff
@@ -75,6 +75,12 @@ func (ch Channel) ValidateBasic() error {
 	return ch.Counterparty.ValidateBasic()
 }
 
+// CloseChannel modifies the Channel with the new State set to CLOSED and FlushStatus set to NOTINFLUSH.
+func CloseChannel(channel *Channel) {
```

Comment:
any reason you went with non-method option? I see we don't generally add mutating methods to these proto-generated types but this feels like it should definitely be one.

Also, and this is personal taste, `types.CloseChannel(channel)` reads oddly in relation to `channel.Close()`.

Reply:
I don't have a strong preference, I guess the main reason would be that the following use looks fine

```golang

channel := NewChannel(...)

channel.Close() 

SetChannel(channel)

```

when it would in fact not do anything (linters would probably actually catch this though)

Correct usage would be

```golang
channel = channel.Close()
```

using a method receiver makes it obvious we're modifying values in-place. If there are preferences on the team I'd be happy to change to whatever people think is the easiest to work with while also being intuitive.

Reply:
wait why not just take it by pointer? 

```go
func (ch *Channel) SetClosed() {
	ch.State = CLOSED
	ch.FlushStatus = NOTINFLUSH
}
```

?

Reply:
the main reason is that we don't really deal with references to channels, e.g. NewChannel returns a struct type, and the SetChannel keeper fn accepts a struct. We would need to do awkward conversions back and forward to persist the changes.

Reply:
I think I'd be happy with a `channel.Close()` in that case.

Reply:
When I first had a quick look at the PR, I also thought that having a method on a pointer receiver would look better, but  then I read online that it seems like generally the consensus ([1](https://go.dev/tour/methods/8), [2](https://yourbasic.org/golang/pointer-vs-value-receiver/)) is to avoid mixing value and pointer receivers?

Reply:
Yeah, in general not a good idea to mix these when also dealing with interfaces since you might have cases where something looks like it implements an interface but in reality doesn't because it has a pointer receiver.

I honestly doubt we'll even have an interface for closing these, seems like it wouldn't be something we'd expose. Current interfaces in `exported` only deal with `get`ting things so we should be fine.

Mixing these can be done relatively safely if done with care, imho.